docker/Dockerfile

```diff
@@ -1,15 +1,27 @@
 FROM python:3.11-slim
+
+WORKDIR /appgate
+USER root
+
 RUN apt-get update && \
-    apt-get install gcc libffi-dev git --yes
-COPY requirements.txt /root/
-RUN python3 -m venv /root/venv && \
-    /root/venv/bin/pip install -r /root/requirements.txt && \
+    apt-get install gcc libffi-dev git --yes && \
+    apt-get clean && \
+    apt-get autoclean
+
+COPY requirements.txt /appgate
+
+RUN python3 -m venv /appgate/venv && \
+    /appgate/venv/bin/pip install --upgrade pip && \
+    /appgate/venv/bin/pip install -r /appgate/requirements.txt && \
     apt-get autoremove gcc --yes && \
```

Comment:
I still think the two `RUN` instructions should be merged into one, or you will be shipping gcc in the actual image. You can move the copy before that.
Speaking of which, what do we need gcc for? Assuming we use only binary wheel python packages it shouldn't be necessary.

Reply:
Looks like it was added a while back in commit 95d5d9786640a3809c34931967d5b8480cb1c1a6. yarl didn't have python3.9 wheel yet back then. I can remove this now.